src/main/java/com/google/cloud/tools/maven/endpoints/framework/OpenApiDocsMojo.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2017 Google Inc. All Right Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.google.cloud.tools.maven.endpoints.framework;
+
+import com.google.api.server.spi.tools.EndpointsTool;
+import com.google.api.server.spi.tools.GetDiscoveryDocAction;
+import com.google.api.server.spi.tools.GetOpenApiDocAction;
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Execute;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * Goal which generates discovery docs
+ */
+@Mojo(name = "openApiDocs", requiresDependencyResolution = ResolutionScope.COMPILE)
+@Execute(phase = LifecyclePhase.PREPARE_PACKAGE)
+public class OpenApiDocsMojo extends AbstractEndpointsWebAppMojo {
+
+  @Parameter(defaultValue = "${project}", readonly = true)
+  private MavenProject project;
+
+  /**
+   * Output directory for discovery docs
+   */
+  @Parameter(defaultValue = "${project.build.directory}/openapi-docs",
+             property = "endpoints.openApiDocDir", required = true)
+  private File openApiDocDir;
+
+  /**
+   * Default hostname of the Endpoint Host.
+   */
+  @Parameter(property = "endpoints.hostname", required = false)
+  private String hostname;
+
+  public void execute() throws MojoExecutionException {
+    try {
+      if (!openApiDocDir.exists()) {
+        if (!openApiDocDir.mkdirs()) {
```

Comment:
I think I need to them to be separate to fail if the directory fails to be created.